nuxeo-runtime/nuxeo-connect-standalone/src/test/java/org/nuxeo/launcher/connect/TestConnectBroker.java

```diff
@@ -21,60 +21,143 @@
+                        byte[] byteArray = FileUtils.readFileToByteArray(pkgZip);
+                        os.write(byteArray, 0, byteArray.length);
+                    }
+                    response.setStatus(HttpServletResponse.SC_OK);
+                } else {
+                    response.setStatus(HttpServletResponse.SC_NOT_FOUND);
+                }
+            }
+        }
+    }
+
     @Before
-    public void setUp() throws IOException {
+    public void beforeEach() throws Exception {
         // set fake Connect connector
-        // TODO The fake connector is not really useful in those tests because all packages are already in the local
-        // store. We must add tests for package requests that need to download packages.
-        String addonJSON = FileUtils.readFileToString(new File("src/test/resources/packages/addon_remote.json"));
-        String hotfixJSON = FileUtils.readFileToString(new File("src/test/resources/packages/hotfix_remote.json"));
-        String studioJSON = FileUtils.readFileToString(new File("src/test/resources/packages/studio_remote.json"));
+        String addonJSON = FileUtils.readFileToString(new File(testStore, "addon_remote.json"));
+        String hotfixJSON = FileUtils.readFileToString(new File(testStore, "hotfix_remote.json"));
+        String studioJSON = FileUtils.readFileToString(new File(testStore, "studio_remote.json"));
         NuxeoConnectClient.getConnectGatewayComponent().setTestConnector(
                 new LocalConnectFakeConnector(addonJSON, hotfixJSON, studioJSON));
 
         // build env
         Environment.setDefault(null);
-        File nuxeoHome = new File("target/launcher");
         FileUtils.deleteQuietly(nuxeoHome);
         nuxeoHome.mkdirs();
         System.setProperty(Environment.NUXEO_HOME, nuxeoHome.getPath());
         System.setProperty(TomcatConfigurator.TOMCAT_HOME, Environment.getDefault().getServerHome().getPath());
 
         // build test packages store
-        File testStore = new File("src/test/resources/packages");
-        FileUtils.copyDirectory(testStore, new File(nuxeoHome, "packages"));
+        buildInitialPackageStore();
+
+        // add fake connect request handler for package downloads
+        server.addHandler(new FakeConnectDownloadHandler());
+    }
+
+    private void buildInitialPackageStore() throws IOException {
+        File nuxeoPackages = new File(nuxeoHome, "packages");
+        File nuxeoStrore = new File(nuxeoPackages, "store");
+        File uninstallFile = new File(testStore, "uninstall.xml");
+        FileUtils.iterateFiles(testStore, new String[] { "zip" }, false).forEachRemaining(pkgZip -> {
+            try {
+                File pkgDir = new File(nuxeoStrore, pkgZip.getName().replace(".zip", ""));
+                ZipUtils.unzip(pkgZip, pkgDir);
+                FileUtils.copyFileToDirectory(uninstallFile, pkgDir);
+            } catch (IOException e) {
+                e.printStackTrace();
```

Comment:
Never use printStackTrace. Here there's no need to catch at all as the method already throws IOException.